packages/runtime/src/store/previews.ts

```diff
@@ -21,18 +21,21 @@ export class PreviewsStore {
     const webcontainer = await webcontainerPromise;
 
     webcontainer.on('port', (port, type, url) => {
-      let previewInfo = this._availablePreviews.get(port);
+      const previewInfos = this._availablePreviews.filter((preview) => preview.port === port);
 
-      if (!previewInfo) {
-        previewInfo = new PreviewInfo(port, type === 'open');
-        this._availablePreviews.set(port, previewInfo);
+      if (previewInfos.length === 0) {
+        const info = new PreviewInfo(port, type === 'open');
+        previewInfos.push(info);
+        this._availablePreviews.push(info);
       }
 
-      previewInfo.ready = type === 'open';
-      previewInfo.baseUrl = url;
+      previewInfos.forEach((info) => {
+        info.ready = type === 'open';
+        info.baseUrl = url;
+      });
 
       if (this._previewsLayout.length === 0) {
-        this.previews.set([previewInfo]);
```

Comment:
This will become:

```ts
this.previews.set(new PreviewInfo(portInfo));
```

Reply:
Applied this, though this required `PreviewInfo` to accept `PortInfo` as constructor argument.